core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SnapshotApiServiceSpec.scala

```diff
@@ -154,4 +154,45 @@ class SnapshotApiServiceSpec extends ApiServiceSpec {
       sealRoute(services.snapshotRoutes) ~>
       check { assertResult(StatusCodes.NotFound) {status} }
   }
+
+  it should "return 204 when a user deletes a snapshot" in withTestDataApiServices { services =>
+    Post(s"${testData.wsName.path}/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "realsnapshot"
+      )
+    )) ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check {
+        val response = responseAs[DataRepoSnapshotReference]
+        assertResult(StatusCodes.Created) {status}
+
+        Delete(s"${testData.wsName.path}/snapshots/${response.referenceId}") ~>
+          sealRoute(services.snapshotRoutes) ~>
+          check { assertResult(StatusCodes.NoContent) {status} }
+
+        //verify that it was deleted
+        Delete(s"${testData.wsName.path}/snapshots/${response.referenceId}") ~>
+          sealRoute(services.snapshotRoutes) ~>
+          check { assertResult(StatusCodes.NotFound) {status} }
+      }
+  }
+
+  it should "return 403 when a user can only read a workspace and tries to delete a snapshot" in withTestDataApiServicesAndUser("reader-access") { services =>
+    Delete(s"${testData.wsName.path}/snapshots/doesntmatter") ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.Forbidden) {status} }
+  }
+
+  it should "return 404 when a user tries to delete a snapshot from a workspace that they don't have access to" in withTestDataApiServicesAndUser("no-access") { services =>
+    Delete(s"${testData.wsName.path}/snapshots/${UUID.randomUUID().toString}") ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.NotFound) {status} }
+  }
```

Comment:
This test would pass (the result would still be NotFound) if the user had access, because the snapshot reference doesn't exist.

Reply:
I was on the fence about that when writing the test. Without inspecting the actual message associated with the 404, there's still no guarantee that the correct 404 is being returned. I'll add the bit to create the snapshot reference first and see if I can also inspect the message to make sure it's the right 404